packages/headless/src/modules/brands/services/index.ts

```diff
@@ -0,0 +1,65 @@
+import { HttpStatus, Injectable } from "@nestjs/common";
+import * as Joi from 'joi';
+import { BrandCreateSchema } from './../validators/brand.create.validator';
+import { Helper } from 'src/helper/helper.interface';
+import { ServiceErrorResponse, ServiceSuccessResponse } from './../../../helper/serviceResponse/service.response.interface';
+import { Brand } from './../../../entity/brand';
+import { BrandRepository } from './../repositories/index';
+import { validateParams } from 'src/decorators/service.validator';
+import { Response } from "express";
+
+@Injectable()
+
+export class Brandservice{
+    constructor( 
+        private brandRepo: BrandRepository,
+        private helper: Helper     
+    ){}
+
+    @validateParams({ schema: BrandCreateSchema})
+    async createBrand(brand: Brand): Promise<ServiceErrorResponse | ServiceSuccessResponse> {
+        const doesBrandExist = await this.brandRepo.findBrandById(brand.id);
+
+        if(doesBrandExist){
```

Comment:
you can refactor code like the bellow
& use isBrandExist instead of doesBrandExist
```
if(isBrandExist) {
    return this.helper.serviceResponse.errorResponse('Brand already exists', null, HttpStatus.BAD_REQUEST);
  }
  const newBrand = await this.brandRepo.createBrand(brand);
  if(!newBrand) return this.helper.serviceResponse.errorResponse("Cant't create brand", null, HttpStatus.BAD_REQUEST,);
  return this.helper.serviceResponse.successResponse(newBrand);
```

Reply:
isBrandExist is not grammartically correct, thats why i renamed it to doesBrandExist

Reply:
it might be... So you can change like bellow
```
if(doesBrandExist) {
    return this.helper.serviceResponse.errorResponse('Brand already exists', null, HttpStatus.BAD_REQUEST);
  }
const newBrand = await this.brandRepo.createBrand(brand);
if(!newBrand) return this.helper.serviceResponse.errorResponse("Cant't create brand", null, HttpStatus.BAD_REQUEST,);
return this.helper.serviceResponse.successResponse(newBrand);
```

Reply:
it might be... So you can change like bellow
```
if(doesBrandExist) {
    return this.helper.serviceResponse.errorResponse('Brand already exists', null, HttpStatus.BAD_REQUEST);
  }
const newBrand = await this.brandRepo.createBrand(brand);
if(!newBrand) return this.helper.serviceResponse.errorResponse("Cant't create brand", null, HttpStatus.BAD_REQUEST,);
return this.helper.serviceResponse.successResponse(newBrand);
```